### ISSUE TITLE
Dynamic style with `css` function

### ISSUE THREAD

**Issue by Ardear** [mui/pigment-css#86]
### Search keywords

css function not work in template

### Latest version

- [X] I have tested the latest version

### Steps to reproduce

1. Clone the example `pigment-css-nextjs-ts` 
2. Create a component as :
```
import { css } from "@pigment-css/react";
import { FunctionComponent } from "react";

interface ButtonProps {
  width: string;
}

export const Button: FunctionComponent<ButtonProps> = ({
  width,
}) => {
  return (
    <div
      className={`${css({
        width,
        height: "fit-content",
        padding: "10px",
        border: "1px solid white",
        borderRadius: 4,
      })}`}
    >
      Button
    </div>
  );
};
```
3. Use the component :
```
// src/app/page.tsx

export default function Home() {
  return (
    <Button width="60px" />
  )
}
```

### Current behavior

Error: `unhandledRejection: This expression cannot be used in the template, because it is a function parameter.`

### Expected behavior

`css` function work fine, and the button id styled `width: 60px`

### Context

I'm trying the pigment-css with nextjs, I want to set some dynamic style, Which style is depend on data, such as: 
```
images.map(({aspectRatio})=> <ImageConponent className={css({aspectRatio})} />)
```
Is it possible to use in server component ?

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by oliviertassinari** [comment]
It sounds like at the very minimum, a lack of docs for this topic, e.g. https://github.com/callstack/linaria/blob/master/docs/DYNAMIC_STYLES.md.

**Comment by brijeshb42** [comment]
True. We need to update documentation around using runtime variables to define styles.

@Ardear This will work if you instead create a `styled` wrapper for the same component:

```tsx
const ButtonWrapped = styled.button<{width: number}>({
  width: (props) => props.width,
  height: "fit-content",
  padding: "10px",
  border: "1px solid white",
  borderRadius: 4,
});

// and use it like this

interface ButtonProps {
  width: string;
}

export const Button: FunctionComponent<ButtonProps> = ({
  width,
}) => {
  return (
    <ButtonWrapped width={width}>
      Button
    </ButtonWrapped>
  );
};
```

Pigment CSS expects most of the css values to be static, one of the reasons to defined styled components or css calls at the file scope, instead of being inside a function.
To apply dynamic width, use the callback signature as shown above. This way, it's able to extract the css at build time while not failing for dynamic values.

**Comment by o-alexandrov** [comment]
You don’t need to move to `styled`.
You can use `css` for `className` prop and [style](https://legacy.reactjs.org/docs/dom-elements.html#style) prop for inline styles.
- styled adds style prop for inline styles in an ambiguous way, whereas by using style prop there’s no ambiguity
